### DISCUSSION TITLE
Cloudflared lxc loses connection to cloudflare after container or host reboot.

### DISCUSSION THREAD

**Discussion by sambarlick** [tteck/Proxmox#1817] [Help Needed]
Hi,

I was wondering if there was a way to store the token from cloudflare in the lxc instead of having to re enter it in the lxc console after every reboot.

**Comment by tteck** [comment]
I've never encountered that issue with cloudflared. I've rebooted many times without any problems.

**Reply by tteck** [reply]
@sambarlick If you run the command `systemctl edit --full cloudflared.service` is there a token shown?
e.g. `ExecStart=/usr/bin/cloudflared --no-autoupdate tunnel run --token eyJhIjoivygdy7qwgsyg3ggsuhhbxzusbuywgudv`

**Reply by sambarlick** [reply]
I rebooted the container and ran that command and it started file not found.

But when I copy the command from cloudflare the tunnel works. But I have to do it every time the container is rebooted.

**Reply by tteck** [reply]
You ran `systemctl edit --full cloudflared.service` in the LXC console and it said file not found?

**Reply by sambarlick** [reply]
Yes. 
![Screenshot_20230920-212936](https://github.com/tteck/Proxmox/assets/47462284/404848dd-6694-454a-a99f-ae3571cd5d61)

**Reply by tteck** [reply]
After inserting your token run this command:

```
cat <<EOF >/etc/systemd/system/cloudflared.service
[Unit]
Description=cloudflared
After=network.target

[Service]
TimeoutStartSec=0
Type=notify
ExecStart=/usr/bin/cloudflared --no-autoupdate tunnel run --token <your token>
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=multi-user.target
EOF
```

**Reply by tteck** [reply]
Then run command `systemctl enable cloudflared.service`

**Comment by sambarlick** [comment]
After inserting the token I'm unable to use the console and enter any command

**Reply by tteck** [reply]
You inserted your token in the snippet I provided? Then executed the snippet in the LXC console?

**Reply by tteck** [reply]
Did you run  the command `systemctl enable cloudflared.service`? Reboot

**Comment by sambarlick** [comment]
Yes and cloudflare is showing the tunnel as healthy.

![Screenshot_20230920-220702](https://github.com/tteck/Proxmox/assets/47462284/887e9f80-c101-4e02-a0c7-03f10b92ce19)

**Reply by tteck** [reply]
Reboot to see if everything is working as intended. You shouldn't have to do anything else but reboot.

**Comment by sambarlick** [comment]
> You ran `systemctl edit --full cloudflared.service` in the LXC console and it said file not found?

Running this command opens the file and the token is stored in the file. But after a reboot cloudflare is showing the tunnel is down.

**Reply by tteck** [reply]
Did you enable the service?

**Reply by tteck** [reply]
`systemctl enable cloudflared.service`

**Comment by sambarlick** [comment]
Yes. I did then rebooted the container.

**Comment by sambarlick** [comment]
I entered the tunnel id instead of the token within the docker command on cloudflare. I changed the file to include the token but it still isn't up after enabling the service and rebooting the container.

**Reply by tteck** [reply]
> docker command

???

I'm not sure about the root cause of this problem, but I recommend starting fresh by setting up a new Cloudflared LXC container. You might find it helpful to refer to [this video](https://www.youtube.com/watch?v=EOcwVjdCAEc&t=117s) for a refresher on how to configure Cloudflare tunnels.

**Reply by tteck** [reply]
But, Debian instead of Docker

**Comment by sambarlick** [comment]
This docker command.

docker run cloudflare/cloudflared:latest tunnel --no-autoupdate run --token eyJhIjo.......

**Reply by tteck** [reply]
The Cloudflared LXC is NOT a Docker installation!

**Comment by sambarlick** [comment]
Running that command every time i reboot is the only thing that establishes a "healthy" tunnel, but then i have to enter it after every reboot.

**Reply by tteck** [reply]
I recommend starting fresh by setting up a new Cloudflared LXC.

**Comment by tteck** [comment]
It seems that the installation you have does not match this repository.

**Comment by sambarlick** [comment]
Ok seems to be working now. I created a new container from your script, created the file, inserted my token and rebooted. Works!!

**Comment by sambarlick** [comment]
Thanks for your help!
